inference-engine/src/vpu/graph_transformer/src/stages/broadcast.cpp

```diff
@@ -60,27 +60,32 @@ class BroadcastStage final : public StageNode {
         VPU_THROW_UNLESS(numOutputs() == 1,
                          "{} stage with name {} must have only 1 output, actually provided {} outputs",
                          type(), name(), numOutputs());
-        if (mode == BroadcastMode::NUMPY) {
-            VPU_THROW_UNLESS(numInputs() == 2,
-                             "{} stage with name {} and numpy mode must have 2 inputs, actually "
+        if (mode == BroadcastMode::EXPLICIT) {
+             VPU_THROW_UNLESS(numInputs() == 3,
+                             "{} stage with name {} and explicit mode must have 3 inputs, actually "
                              "provided {} inputs", type(), name(), numInputs());
             assertInputsOutputsTypes(this,
-                                     {{dataPrecision}, {DataType::S32}},
+                                     {{dataPrecision}, {DataType::S32}, {DataType::S32}},
                                      {{dataPrecision}});
-
         } else {
-            VPU_THROW_UNLESS(numInputs() == 3,
-                             "{} stage with name {} and explicit mode must have 3 inputs, actually "
+             VPU_THROW_UNLESS(numInputs() == 2,
+                             "{} stage with name {} and numpy or bidirectional mode must have 2 inputs, actually "
                              "provided {} inputs", type(), name(), numInputs());
             assertInputsOutputsTypes(this,
-                                     {{dataPrecision}, {DataType::S32}, {DataType::S32}},
+                                     {{dataPrecision}, {DataType::S32}},
                                      {{dataPrecision}});
         }
     }
 
     void serializeParamsImpl(BlobSerializer& serializer) const override {
         const auto mode = attrs().getOrDefault<BroadcastMode>("mode", BroadcastMode::NUMPY);
-        serializer.append(static_cast<uint32_t>(mode == BroadcastMode::NUMPY ? 0 : 1));
+        if (mode == BroadcastMode::NUMPY) {
```

Comment:
I guess
`serializer.append(static_cast<BroadcastMode>(mode));`
should also work.